src/ucs/config/parser.c

```diff
@@ -379,6 +379,145 @@ int ucs_config_sprintf_time(char *buf, size_t max, void *src, const void *arg)
     return 1;
 }
 
+int ucs_config_sscanf_speed(const char *buf, void *dest, const void *arg)
+{
+    size_t *dst     = (size_t*)dest;
+    char    str[16] = {0};
+    int     offset  = 0;
+    size_t  divider;
+    size_t  units;
+    double  value;
+    int     num_fields;
+
+    num_fields = sscanf(buf, "%lf%16s", &value, str);
+    if (num_fields < 2) {
+        return 0; /* bytes or bits? */
+    }
+
+    ucs_assert(num_fields == 2);
+
+    switch (str[offset]) {
+    case 'T':
+        units = UCS_TBYTE;
+        break;
+    case 'G':
+        units = UCS_GBYTE;
+        break;
+    case 'M':
+        units = UCS_MBYTE;
+        break;
+    case 'K':
+        units = UCS_KBYTE;
+        break;
+    case 'B':
+    case 'b':
+        units = 1;
+        goto bytes;
+    default:
+        return 0;
+    }
+
+    offset++;
+
+bytes:
+    switch (str[offset]) {
+    case 'B':
+        divider = 1;
+        break;
+    case 'b':
+        divider = 8;
+        break;
+    default:
+        return 0;
+    }
+
+    offset++;
+
+    switch (str[offset]) {
```

Comment:
can we just check that the remainder (str + offset) is either "ps" or "/s" ? with strcmp